api/v1alpha1/nodepool_types.go

```diff
@@ -57,6 +68,17 @@ type NodePoolList struct {
 	Items           []NodePool `json:"items"`
 }
 
+type NodePoolManagement struct {
+	// +kubebuilder:default=true
+	AutoUpgrade bool `json:"autoUpgrade"`
```

Comment:
Playing into the same flow, is this a gate that would prevent reconciliation of the nodes to a new version?

Reply:
I propose we remove this field and start a followup discussion about upgrade policies.

Reply:
The intention was to default this to true and start by only implementing that. I can drop it for now if you prefer.

Reply:
I would propose describing the behavior (not a "default" per say because there's no option) and then starting a new discussion PR around upgrade policies — keeps everything simple for now and gives us a space to go deeper on this part